### ISSUE TITLE
Coin was transferred out automatically

### ISSUE THREAD

**Issue by stlobe** [Snipa22/nodejs-pool#361]
I got a very strange problem in my pool. The block was found and coins were deposited to my wallet. But I don't know why all coins were transferred out to other (unknown) after few hours later. Can anyone assist??

**Comment by bobbieltd** [comment]
Got hacked ?

**Comment by stlobe** [comment]
I don't think so. I find all coins for that block were transferred. It is Solo type but I can't find any coin in Solo wallet.

**Comment by bobbieltd** [comment]
Let me guess ! It’s not Monero or Monero forks. A kind of forknote shits ! 
Show transfers. Do you see out coming transaction on pool wallet ? Or it just disappears.

**Comment by stlobe** [comment]
Yes I see that transaction and got hash ID as well. Originally, I just guess coins were transferred out to Solo wallet because it is pool function. But I waited more than 1 day......no coin found :(

**Comment by bobbieltd** [comment]
Did yu check if that transaction hash is correct on public blockchain explorer ? If yes, the pool (payments) or someone (not you) send coins out.

**Comment by stlobe** [comment]
Yes I also checked it. It is correct. But I am the one can transfer coins out by manual. That's why I guess pool transfer out the coins automatically.

**Comment by stlobe** [comment]
But I can't find any payment was occurred in log file.....

**Comment by bobbieltd** [comment]
So the pool automatically transfers coins out to unknown address ? 
Did you check the balance table in mysql ? Pool only transfers coins to addresses in the balance table ?

**Comment by stlobe** [comment]
Yes checked also. The Solo address, developer's address and Sniper;'s address were found in it.
I am not sure the coins were transferred to developer and sniper or not.....

**Comment by stlobe** [comment]
the solo address can't receive its coins.

**Comment by bobbieltd** [comment]
By using wallet CLI, you can list_transfers or show_tranfers to trace back that unkown address. Unknown here means you don’t know address or you don’t know this address belonging to who ?

**Comment by stlobe** [comment]
I just got hash ID but I don't know how to trace back the unknown address. by using cli. Can you tell me more?

**Comment by bobbieltd** [comment]
Log in your pool wallet by command line (ask Google how to with your coin name).
Type help to show all commands and look for the comand that will show all transfers ( I guess list_tranfers ).
It will list all transactions in pool wallet and you can see the wallet address that the pool send the coins out.

**Comment by bobbieltd** [comment]
Running simplewallet without parameters. I wonder how you created pool/fee wallet and runned wallet RPC. Someone else did for you ?

**Comment by stlobe** [comment]
I tried but can't release the address.....

**Comment by bobbieltd** [comment]
Post here the screenshot of all pool wallet tranfers from command line if there are not any things private here.

**Comment by bobbieltd** [comment]
Here is the point. You can not blame the pool transfering coins to other (you still don’t know what is the sent outgoing address.
If you don’t know how to deal with wallet CLI, now it’s the time to practice it.
Running pool is not setup and forget.
For your unreceived coin, it can be stuck transaction. It means the pool sent you coins but that transaction is still stuck on blockchain. Or you didn’t fully synchronize your wallet when you check your personal (solo) wallet.
If you can not give more details, no one can know what happened to you.

**Comment by stlobe** [comment]
nono I am not blaming the pool. I just feel strange on my issue. 
Here are the transaction of this block.
 1526986     in       2018-03-11       6.297466147479 43463f4360e72d6d90c5fe602902eb94de64cddd34e4ecd548961b4e222c6b5e 0000000000000000 -
 1527161    out       2018-03-11       6.358359288989 db6f2f9a4a3f20f94ff2aff3b04415a5ff6e69433bbc77e3ef1754faec3bb228 0000000000000000 0.011260480000

**Comment by bobbieltd** [comment]
Is it output of list_transfers ?. What command did you use (list_transfers, show_transfers ?) What is your coin ?

**Comment by bobbieltd** [comment]
Oufff, it’s Monero. I admire you. Command is show_transfers.

**Comment by bobbieltd** [comment]
To see outgoing addresses, you need to access wallet on the pool servers (it is cached on servers, don’t delete anything before understanding what happened ). If you access wallet outside, you can’t see any addresses.

**Comment by stlobe** [comment]
I accessed to pool server and cant find any payment inside. I guess I missed the command to retrieve the address. Can anyone assist?

**Comment by bobbieltd** [comment]
In case you get lost old wallet data, you can try to track that transaction with prove sending.
https://getmonero.org/resources/user-guides/prove-payment.html

Get tx_key from wallet CLI. Then you test prove sending with your solo address, dev address, Snipa address. All addresses on balance table should be checked.
I hope the pool sent to you.

**Comment by stlobe** [comment]
Thank you I will try later.

**Comment by bobbieltd** [comment]
It’s quite a lot if money. You can check prove sending if the pool send that amount to your wallet address to avoid worriness in waiting.
If prove sending confirms that transaction is to your solo address. There is nothing to worry.

**Comment by stlobe** [comment]
Already tried but no fx key can be found.... And no coin was transferred to solo wallet. Very very strange

**Comment by bobbieltd** [comment]
Sorry, it seems you don’t have data at pool server to trace this (no address destination, no tx key). It’s quite deadlock. Monero is privacy currency. We need another approach.
That outgoing transaction is done two days ago.
Here are my assumption : You got hacked. Someone can access your server to send coins out and remove traces at wallet data.

It’s quite a big money. I hope that you will receive it by some how.

**Comment by bobbieltd** [comment]
One more question, what do you see the amount total due of your wallet address at pool now ? Is there any payment done on website ?

**Comment by bobbieltd** [comment]
Put your wallet address (solo) at Dashboard website and give me your Total due and Total paid.

**Comment by stlobe** [comment]
No payment is done on website. I checked the server log and firewall log as well. No hacker accessed between receiving and sending coins.

**Comment by bobbieltd** [comment]
Your infos that you gave is so limited. 
If you want to trace the issue, you need to provide the most comprehensive info as possible. It’s vague and contradictory now.
1st block ? Total due ? Total paid ?
Here is the thing. If the pool send coins out, you can see destination address and can get tx key with pool wallet on pool server. If you can not, it’s not the pool that sent coins out.

**Comment by stlobe** [comment]
This is the first block for this pool. No similar problem was occurred in other pools. That's why I think this is not hacker problem. I am still checking other information that I can find in this pool server.

**Comment by stlobe** [comment]
Hi Snipa, Any idea?

**Comment by MoneroOcean** [comment]
Maybe you got orphan block (so its reward was cancelled)? Just a wild guess.

**Comment by TwixtedChaox** [comment]
It's not just you. This has happened to my pool SEVERAL times (I stopped running snipa pool and problem stopped...) currently running my own variant of snipas pool software as the coin "flush" happens approx 12 hours after confirmed block. I am still trying to locate where in the software it flushes the coins because I have had over 50 XMR stolen that way

**Comment by bobbieltd** [comment]
50 XMR stolen !?! LOL , the source code is open. If the pool flush the coins, some one will notice long time ago.

**Comment by stlobe** [comment]
Over 50XMR were stolen? I stopped to find it and set up a new pool again.
